File: 7. Classes Abstratas, Sobrescrita e Polimorfismo/Investimentos (Java) /Investimento.java
public abstract class Investimento {
    private String nome;
    protected double valor_inicial;

    public Investimento(String nome, double valor_inicial) {
        this.nome = nome;
        this.valor_inicial = valor_inicial;
    }

    public String getNome() {
        return this.nome;
    }

    public double getValor_inicial() {
        return this.valor_inicial;
    }

    public abstract double retorno();
}

File: 7. Classes Abstratas-Sobrescrita e Polimorfismo/Investimentos (C++)/Investimento.cpp
#include "Investimento.hpp"

Investimento::Investimento(string n, double in) : nome(n), investimento_inicial(in) {}

Investimento::~Investimento() {}

string Investimento::get_nome() const {
    return nome;
}

double Investimento::get_investimento_inicial() const {
    return investimento_inicial;
}